### PR TITLE
Docstring cleanup with new DQ and return system

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
We switched default.qubit to the new device API implementation, and some docstrings need updating.

**Description of the Change:**
Make all docstrings reflect what actually happens. Some notes:
- Snapshot has exposed that states may be floats instead of complex before being returned, though this does not affect any results. I'll add a story to track our decision on this
- `_grad.py` needed updating for the jacobian docs because `qml.jacobian` doesn't work with >1 MP being returned, so I return `qml.probs()` to get measurements that increase the returned dimension
- made other changes like tracker and batch_execute to use the new device API

**Benefits:**
The docstrings are correct!

**Possible Drawbacks:**
- Ugly PR!
- I made quite a few changes about the interface returned, but I left a few out. I think there's more work to be done here, like how we handle no-arg QNodes and the interface returned there. It's always autograd instead of None, but that's not the worst of it

### CODE REVIEW

File: pennylane/gradients/__init__.py [hunk 1]
```diff
@@ -149,8 +149,9 @@ def circuit(weights):
 >>> circuit(weights)
 tensor([0.9658079, 0.0341921], requires_grad=True)
 >>> qml.gradients.param_shift(circuit)(weights)
-tensor([[-0.04673668, -0.09442394, -0.14409127],
-        [ 0.04673668,  0.09442394,  0.14409127]], requires_grad=True)
+(tensor([-0.04673668,  0.04673668], requires_grad=True),
+ tensor([-0.09442394,  0.09442394], requires_grad=True),
+ tensor([-0.14409127,  0.14409127], requires_grad=True))
```
Comment (timmysilv) [inline pennylane/gradients/__init__.py:154]: I don't know why the shape changed here. something to look into?

File: pennylane/ops/qubit/observables.py [hunk 2]
```diff
@@ -574,14 +574,14 @@ def label(self, decimals=None, base_label=None, cache=None):
         **Example:**
 
         >>> state_vector = np.array([0, 1, 1, 0])/np.sqrt(2)
-        >>> StateVectorProjector(state_vector, wires=(0, 1)).label()
+        >>> qml.Projector(state_vector, wires=(0, 1)).label()
```
Comment (timmysilv) [inline pennylane/ops/qubit/observables.py:577]: it can't be directly constructed because the `__new__` override doesn't take args. could fix it, but i'd like to only change docs as much as possible in this PR

File: pennylane/qchem/tapering.py [hunk 3]
```diff
@@ -468,7 +468,7 @@ def taper_hf(generators, paulixops, paulix_sector, num_electrons, num_wires):
     >>> symbols = ['He', 'H']
     >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]])
     >>> mol = qml.qchem.Molecule(symbols, geometry, charge=1)
-    >>> H, n_qubits = qml.qchem.molecular_hamiltonian(symbols, geometry)
+    >>> H, n_qubits = qml.qchem.molecular_hamiltonian(symbols, geometry, charge=1)
```
Comment (timmysilv) [inline pennylane/qchem/tapering.py:471]: fails without this

File: pennylane/qinfo/transforms.py [hunk 4]
```diff
@@ -839,7 +838,7 @@ def circuit_ry(x):
             qml.RY(x, wires=0)
             return qml.state()
 
-    >>> qml.qinfo.fidelity(circuit_rx, circuit_ry, wires0=[0], wires1=[0])((), (0.2))
+    >>> qml.qinfo.fidelity(circuit_rx, circuit_ry, wires0=[0], wires1=[0])(None, (0.2))
```
Comment (timmysilv) [inline pennylane/qinfo/transforms.py:841]: fails with the empty tuple

File: pennylane/templates/subroutines/hilbert_schmidt.py [hunk 5]
```diff
@@ -268,9 +268,10 @@ def compute_decomposition(
         )
 
         # Unitary U
-        for op_u in u_tape.operations:
-            qml.apply(op_u)
-            decomp_ops.append(op_u)
+        if qml.QueuingManager.recording():
+            decomp_ops.extend(qml.apply(op_u) for op_u in u_tape.operations)
+        else:
+            decomp_ops.extend(u_tape.operations)
```
Comment (timmysilv) [inline pennylane/templates/subroutines/hilbert_schmidt.py:274]: real change alert! I can undo this here if needed, but the docstring failed because we tried to apply outside of a context

File: pennylane/transforms/zx/converter.py [hunk 6]
```diff
@@ -337,7 +337,7 @@ def processing_fn(res):
             else:
                 expanded_operations.append(op)
 
-        expanded_tape = QuantumScript(expanded_operations, mapped_tape.measurements, [])
+        expanded_tape = QuantumScript(expanded_operations, mapped_tape.measurements)
```
Comment (timmysilv) [inline pennylane/transforms/zx/converter.py:340]: technically a real change, just removed the empty `prep` to avoid raising a warning

File: pennylane/transforms/tape_expand.py [hunk 7]
```diff
@@ -467,7 +467,7 @@ def circuit():
     Now let's set up a context where the custom decomposition will be applied:
 
     >>> with qml.transforms.set_decomposition({qml.CNOT : custom_cnot}, dev):
-    ...     print(qml.draw(circuit)())
+    ...     print(qml.draw(circuit, wire_order=[0, 1])())
```
Comment (timmysilv) [inline pennylane/transforms/tape_expand.py:470]: with DQL, this stayed 0-1 regardless. with new DQ, it swaps them because `1` comes first in the tape. regression in drawing behaviour?

**Review by timmysilv — COMMENTED** [review]
some inline comments to help reviewers. 2 are real changes

File: pennylane/devices/default_qubit.py [hunk 8]
```diff
@@ -12,7 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-This module contains the next generation successor to default qubit
+The default.qubit device is PennyLane's standard qubit-based device.
+
+This module contains the next generation successor to default qubit.
```
Comment (albi3ro) [inline pennylane/devices/default_qubit.py:17]:
```suggestion
```

default qubit is a successor to default qubit? Don't think we need this sentence anymore.

File: pennylane/gradients/__init__.py (hunk 1 above)
Comment (albi3ro) [reply pennylane/gradients/__init__.py:154]: this looks like another return types change.

File: pennylane/devices/default_qubit.py [hunk 9]
```diff
@@ -12,7 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """
-This module contains the next generation successor to default qubit
+The default.qubit device is PennyLane's standard qubit-based device.
+
```
Comment (timmysilv) [inline pennylane/devices/default_qubit.py:16]:
```suggestion
```

File: pennylane/devices/device_api.py
Comment (timmysilv) [inline pennylane/devices/device_api.py:1]: doc check action failed without this saying it was ambiguous which Device class was being referenced. now it passes, so I'm happy

File: pennylane/qchem/tapering.py [hunk 10]
```diff
@@ -650,14 +650,14 @@ def taper_operation(
...
-    >>> H, n_qubits = qchem.molecular_hamiltonian(symbols, geometry)
+    >>> H, n_qubits = qchem.molecular_hamiltonian(symbols, geometry, charge=1)
     >>> generators = qchem.symmetry_generators(H)
     >>> paulixops = qchem.paulix_ops(generators, n_qubits)
     >>> paulix_sector = qchem.optimal_sector(H, generators, mol.n_electrons)
     >>> tap_op = qchem.taper_operation(qml.SingleExcitation, generators, paulixops,
     ...                                paulix_sector, wire_order=H.wires, op_wires=[0, 2])
     >>> tap_op(3.14159)
-    [Exp(1.570795j PauliY)]
+    [Exp(1.5707949999999993j PauliY), Exp(0j Identity)]
```
Comment (albi3ro) [inline pennylane/qchem/tapering.py:660]: Looks like a bunch more spots in this docstring are showing the same mistake.

We might also want to double-check with @soranjh that this is the behaviour we expect.

File: pennylane/templates/subroutines/hilbert_schmidt.py (hunk 5 above)
Comment (albi3ro) [reply pennylane/templates/subroutines/hilbert_schmidt.py:274]: I'm breaking this change into a separate PR so we can link it as evidence in the technical roadmap.

File: pennylane/qchem/tapering.py (hunk 10 above)
Comment (soranjh) [reply pennylane/qchem/tapering.py:660]: The new print looks strange. Not sure what causes this but we can take a look.

File: pennylane/transforms/tape_expand.py (hunk 7 above)
Comment (albi3ro) [reply pennylane/transforms/tape_expand.py:470]: We can update `drawer/draw.py:L262` to be:
```
            _wire_order = wire_order or qnode.device.wires or qnode.tape.wires
```

Which I think would actually be the better behavior.  Want me to open up a quick fix for that?

Comment (timmysilv) [reply pennylane/transforms/tape_expand.py:470]: sure!

File: pennylane/measurements/classical_shadow.py [hunk 11]
```diff
@@ -195,10 +195,10 @@ def circuit():
             dev = qml.device("default.qubit", wires=2, shots=5)
 
             measurements1 = [qml.classical_shadow(wires=(0,1), seed=10)]
-            tape1 = qml.tape.QuantumTape(ops, measurements1)
+            tape1 = qml.tape.QuantumTape(ops, measurements1, shots=5)
```
Comment (lillian542) [inline pennylane/measurements/classical_shadow.py:198]: should `shots=5` be removed from the device above then?

File: pennylane/templates/subroutines/hilbert_schmidt.py [hunk 12]
```diff
@@ -248,6 +248,7 @@ def cost_lhst(parameters, v_function, v_wires, u_tape):
 
         Now that the cost function has been defined it can be called for specific parameters:
 
+
```
Comment (timmysilv) [inline pennylane/templates/subroutines/hilbert_schmidt.py:251]:
```suggestion
```

File: pennylane/templates/subroutines/hilbert_schmidt.py (hunk 5 above)
Comment (timmysilv) [reply pennylane/templates/subroutines/hilbert_schmidt.py]: #4719 merged! file is now untouched in this PR

**Review by lillian542 — COMMENTED** [review]
Would this be an appropriate PR for updating the `qml.device` docstring as well?

**Review by albi3ro — APPROVED** [review]
Thanks for tracking all these down 🚀

File: pennylane/gradients/spsa_gradient.py [hunk 13]
```diff
@@ -241,7 +241,7 @@ def spsa_grad(
         device evaluation. Instead, the processed tapes, and post-processing
         function, which together define the gradient are directly returned:
 
-        >>> ops = [qml.RX(p, wires=0) for p in params]
+        >>> ops = [qml.RX(params[0], 0), qml.RY(params[1], 0), qml.RX(params[2], 0)]
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/gradients/spsa_gradient.py:244]: The results below corresponded to these operations, not the operations stated.

[the remark this answers is not in the document]
Comment (timmysilv) [reply pennylane/gradients/spsa_gradient.py:244]: the results are actually with the RY in the middle. I wasn't getting those, which is when I asked Romain and he scrolled up and realized the circuit-version is XYX, not XXX. after updating to XYX, I re-ran and confirmed I got the results here.

File: pennylane/measurements/classical_shadow.py (hunk 11 above)
Comment (timmysilv) [reply pennylane/measurements/classical_shadow.py:198]: I remember I tried it once a while back and it caused an issue with some corner case. I figure it doesn't hurt to just leave it, plus users expect to have to give shots to the device.

**Comment by timmysilv** [comment]
> Would this be an appropriate PR for updating the `qml.device` docstring as well?

@lillian542 that would be wise to update... but maybe I'll do it in another PR - might merit more discussion. That docstring pre-dates `default.qubit.jax` and `default.qubit.torch`!

**Review by lillian542 — APPROVED** [review]
Separate PR for `qml.device` sounds good! Have an approval 🚀

**Comment by timmysilv** [comment]
[sc-44405]
